File: _posts/Programmers-main/프로그래머스/lv1/숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = ''
    x_dic, y_dic = dict(), dict()
    for i in X:
        if i not in x_dic:
            x_dic[i] = 1
        else:
            x_dic[i] += 1
    for j in Y:
        if j not in y_dic:
            y_dic[j] = 1
        else:
            y_dic[j] += 1
    for key in x_dic.keys():
        if key in y_dic:
            answer += key * min(x_dic[key], y_dic[key])
    answer = "".join(sorted(answer, reverse = True))
    if answer == "":
        answer = "-1"
    elif answer.count("0") == len(answer):
        answer = "0"
    return answer

File: _posts/Programmers_cpp-main/프로그래머스/lv1/숫자 짝꿍/숫자 짝꿍.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string X, string Y) {
    string answer = "";
    for (int i = 9; i >= 0; i--)
    {
        int X_count = count(X.begin(), X.end(), char(i+48));
        int Y_count = count(Y.begin(), Y.end(), char(i+48));
        if (X_count > Y_count) 
        {
            for (int j = 0; j < Y_count; j++) answer += to_string(i);
        }
        else
        {
            for (int j = 0; j < X_count; j++) answer += to_string(i);
        }
    }
    if (answer[0] == '0') 
    {
        answer = "0";
        return answer;
    }
    if (answer == "") answer = "-1";
    return answer;
}